File: leetCode/fukun/101-200/lc_108.js
function TreeNode(val) {
    this.val = val;
    this.left = this.right = null;
}

const sortedArrayToBST = function (nums) {
    const doGrow = function (l, r) {
        if (l > r) return null;
        let cut = l + Math.floor((r - l) / 2);
        let root = new TreeNode(nums[cut]);
        root.left = doGrow(l, cut - 1);
        root.right = doGrow(cut + 1, r);
        return root;
    };

    return doGrow(0, nums.length - 1);
};


console.log(sortedArrayToBST([-10, -3, 0, 5, 9]));

File: leetCode/XSS/lc_108.java
import java.util.Arrays;

public class lc_108 {
    public static void main(String[] args) {

    }
    public static class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode(int x) { val = x; }
    }
    static class Solution {
        public TreeNode sortedArrayToBST(int[] nums) {
            int n=nums.length;
            if(n==0)return null;
            int mid=n/2;
            TreeNode root=new TreeNode(nums[mid]);
            if(mid>0) root.left=sortedArrayToBST(Arrays.copyOfRange(nums,0,mid));
            else root.left=null;
            if(mid<n-1)root.right=sortedArrayToBST(Arrays.copyOfRange(nums,mid+1,n));
            else root.right=null;
            return root;
        }
    }
}
